src/librustdoc/lib.rs

```diff
@@ -429,24 +421,29 @@ fn main_args(args: &[String]) -> i32 {
             early_error(ErrorOutputType::default(), &err.to_string());
         }
     };
+
+    // Note that we discard any distinction between different non-zero exit
+    // codes from `from_matches` here.
     let options = match config::Options::from_matches(&matches) {
         Ok(opts) => opts,
-        Err(code) => return code,
+        Err(code) => return if code == 0 { Ok(()) } else { Err(()) },
```

Comment:
I think this should return `Err(ErrorReported)` so that the caller doesn't have to do the `== Ok(())` dance.